webhook_server/web/log_viewer.py

```
@@ -554,34 +578,171 @@ async def get_workflow_steps_json(self, hook_id: str) -> dict[str, Any]:
+
+        # Extract timing info
+        start_time = timing["started_at"]
+        total_duration_ms = timing["duration_ms"]
+
+        # Transform workflow_steps dict to array format
+        # Sort by timestamp to maintain execution order
+        steps_list = []
+        for step_name, step_data in workflow_steps.items():
+            # Validate step_data is a dict before accessing its fields
+            if not isinstance(step_data, dict):
+                raise ValueError(f"malformed log: step_data for {step_name} is not a dict")
```

Ruff (0.14.13)

Prefer `TypeError` exception for invalid type

(TRY004)